### PR TITLE
[PATCH v6] changelog: updates for v1.18.0.0

### PR DESCRIPTION
**Pull request by Bill-Fischofer-Linaro**
Add updates for v1.18.0.0 (Tiger Moth RC2)

Signed-off-by: Bill Fischofer <bill.fischofer@linaro.org>

### CODE REVIEW

**Comment by Bill-Fischofer-Linaro** [comment]
Posted for initial review. Do not merge yet. I will update this after PR #497 and any other PRs that are needed for RC2 are merged.

File: CHANGELOG [hunk 1]
```diff
@@ -1,3 +1,217 @@
...
+inbound packet stream. For lookaside processing they are always visible. For TX
+processing, the `odp_ipsec_out_opt_t` struct specifies the `tfc_pad` bit if
+the packet is to be padded or the `tfc_dummy` bit if a dummy packet is to
+be inserted. The dummy packet length is specified by the `tfc_pad_len` option.
+
+==== Pktio null device support
+The `odp_pktio_open()` API may now specify devices of class `null` to indicate
+the PktIO is treated as a dummy device. Null devices behave just like Linux
+`/dev/null` in that they never receive packets and simply discard any packets
+sent to them.
```
Comment (lumag) [inline CHANGELOG]: It should be noted, that there are/might be implementations that do not provide null PktIO

File: CHANGELOG [hunk 2]
```diff
@@ -1,3 +1,217 @@
...
+A number of crypto refinements are included in this release:
+
+* The single initialization vector (`iv`) in the `odp_crypto_session_param_t`
+is replaced by a separate `cipher_iv` and `auth_iv` fields.
+
+* The single initialization vector (`override_iv_ptr`) in the
+`odp_crypto_op_param_t` is replaced by a separate `cipher_iv_ptr` and
+`auth_iv_ptr` fields.
+
+* The special nature of GCM and GMAC cipher and authentication algorithms is
```
Comment (lumag) [inline CHANGELOG]: Here and below. GCM, CCM, ChaCha20-Poly1305 are not "cipher and authentication" algorithms. They are "authenticated encryption" modes/algorithms.

File: CHANGELOG [hunk 3]
```diff
@@ -1,3 +1,217 @@
...
+ODP exposes symbols outside of odp*/_odp* namespace
+
+==== https://bugs.linaro.org/show_bug.cgi?id=3210[Bug 3210]
+packet header parsing routines should verify header checksums
+
+==== https://bugs.linaro.org/show_bug.cgi?id=3245[Bug 3245]
+Cannot run l2fwd application on Cavium ThunderX platform
+
+==== https://bugs.linaro.org/show_bug.cgi?id=3517[Bug 3517]
+timer test might fail
```
Comment (lumag) [inline CHANGELOG]: Must be fixed before RC2.

File: CHANGELOG [hunk 4]
```diff
@@ -1,3 +1,217 @@
...
+
+==== https://bugs.linaro.org/show_bug.cgi?id=3597[Bug 3597]
+new generator test assumes that null:0 pktio is always present
```
Comment (lumag) [inline CHANGELOG]: Is fixed already

File: CHANGELOG [hunk 5]
```diff
@@ -1,3 +1,217 @@
...
+
+==== https://bugs.linaro.org/show_bug.cgi?id=3602[Bug 3602]
+validation: cls: capability field is not checked for test cases
+
+==== https://bugs.linaro.org/show_bug.cgi?id=3611[Bug 3611]
+ODP linux generic fails on AArch64 in non-ABI-compat mode
+
+==== https://bugs.linaro.org/show_bug.cgi?id=3628[Bug 3628]
+Another timer_main failure
```
Comment (lumag) [inline CHANGELOG]: Should be fixed before RC2.

File: CHANGELOG (hunk 3 above)
Comment (Bill-Fischofer-Linaro) [reply CHANGELOG]: As discussed today, this must be fixed before Tiger Moth release. RC2 is just the API freeze level.

File: CHANGELOG (hunk 5 above)
Comment (Bill-Fischofer-Linaro) [reply CHANGELOG]: Again, should be fixed before Tiger Moth release. RC2 is API freeze level, so this doesn't affect any APIs.

**Comment by Bill-Fischofer-Linaro** [comment]
@muvarov I've reposted this PR but for some reason it's not being identified as v2. This is actually v2. Changes:

* Rebased to latest `next` branch
* Incorporate review comments by @lumag 
* Pick up latest API change:`odp_packet_data_seg_len()`

I v3 will follow depending on what other PRs get merged before RC2 freezes.

**Comment by Bill-Fischofer-Linaro** [comment]
Changes in v4:

* Added selectable scheduler and ring-based queues (PRs to be part of RC2)
* Updated bug list for recently merged fixes

File: CHANGELOG (hunk 5 above)
Comment (muvarov) [reply CHANGELOG]: Bug 3628 fixed, line has to be removed.

File: CHANGELOG [hunk 6]
```diff
@@ -1,3 +1,248 @@
...
+==== https://bugs.linaro.org/show_bug.cgi?id=3210[Bug 3210]
+packet header parsing routines should verify header checksums
+
+==== https://bugs.linaro.org/show_bug.cgi?id=3245[Bug 3245]
+Cannot run l2fwd application on Cavium ThunderX platform
+
+==== https://bugs.linaro.org/show_bug.cgi?id=3602[Bug 3602]
+validation: cls: capability field is not checked for test cases
+
+==== https://bugs.linaro.org/show_bug.cgi?id=3611[Bug 3611]
```
Comment (muvarov) [inline CHANGELOG]: 3611 fixed, has to be removed. @lumag please confirm from your side.

File: CHANGELOG [hunk 7]
```diff
@@ -1,3 +1,248 @@
...
+==== https://bugs.linaro.org/show_bug.cgi?id=2988[Bug 2988]
+ODP exposes symbols outside of odp*/_odp* namespace
+
+==== https://bugs.linaro.org/show_bug.cgi?id=3210[Bug 3210]
+packet header parsing routines should verify header checksums
+
+==== https://bugs.linaro.org/show_bug.cgi?id=3245[Bug 3245]
+Cannot run l2fwd application on Cavium ThunderX platform
+
+==== https://bugs.linaro.org/show_bug.cgi?id=3602[Bug 3602]
```
Comment (muvarov) [inline CHANGELOG]: Bug 3602 is not a but. It's just remided to Bala. Entry in changelog has to be removed.

File: CHANGELOG [hunk 8]
```diff
@@ -1,3 +1,248 @@
...
+
+=== Known Issues
+
+==== https://bugs.linaro.org/show_bug.cgi?id=2988[Bug 2988]
+ODP exposes symbols outside of odp*/_odp* namespace
+
+==== https://bugs.linaro.org/show_bug.cgi?id=3210[Bug 3210]
+packet header parsing routines should verify header checksums
+
+==== https://bugs.linaro.org/show_bug.cgi?id=3245[Bug 3245]
```
Comment (muvarov) [inline CHANGELOG]: Bug 3245 is not related to linux-generic. Has to be remove from changelog.

File: CHANGELOG [hunk 9]
```diff
@@ -1,3 +1,248 @@
...
+==== https://bugs.linaro.org/show_bug.cgi?id=3632[Bug 3632]
+Creating a pool with total size over 4.29GB (UINT32_MAX) leads to
+undefined behavior
+
+=== Known Issues
+
+==== https://bugs.linaro.org/show_bug.cgi?id=2988[Bug 2988]
+ODP exposes symbols outside of odp*/_odp* namespace
+
+==== https://bugs.linaro.org/show_bug.cgi?id=3210[Bug 3210]
```
Comment (muvarov) [inline CHANGELOG]: Bug 3210 looks like improvement not a bug. I would remove it from changelog also.

**Comment by Bill-Fischofer-Linaro** [comment]
Changes for v5:

* Removed references to bugs 3210, 3245, 3602, and 3611 as requested.

File: CHANGELOG [hunk 10]
```diff
@@ -1,3 +1,236 @@
...
+Creating a pool with total size over 4.29GB (UINT32_MAX) leads to
+undefined behavior
+
+=== Known Issues
+
+==== https://bugs.linaro.org/show_bug.cgi?id=2988[Bug 2988]
+ODP exposes symbols outside of odp*/_odp* namespace
+
+==== https://bugs.linaro.org/show_bug.cgi?id=3628[Bug 3628]
+Another timer_main failure
```
Comment (muvarov) [inline CHANGELOG]: timer is merged.  https://bugs.linaro.org/show_bug.cgi?id=3628

**Comment by Bill-Fischofer-Linaro** [comment]
Changes for v6:

* Restored Bug 3611 as fixed
* Moved Bug  3628 to fixed

**Comment by muvarov** [comment]
Merged.

File: CHANGELOG (hunk 6 above)
Comment (lumag) [reply CHANGELOG]: No, 3611 is not fixed.

**Comment by lumag** [comment]
@muvarov @Bill-Fischofer-Linaro 3611 is not fixed, so changelog should be corrected.

**Comment by Bill-Fischofer-Linaro** [comment]
PR #500 created to correct Changelog for v1.18.0.0.

**Pull request closed without merging**